components/base/kinematicbase/ptgKinematics.go

```diff
@@ -139,12 +140,30 @@ func wrapWithPTGKinematics(
 		origin = originPIF.Pose()
 	}
 	startingState := baseState{currentInputs: zeroInput}
+	// construct localization frame
+	localizationFrame, err := referenceframe.NewPoseFrame(
+		b.Name().ShortName()+"LocalizationFrame",
+		[]referenceframe.Limit{
+			{Min: math.Inf(-1), Max: math.Inf(1)},
+			{Min: math.Inf(-1), Max: math.Inf(1)},
+			{Min: math.Inf(-1), Max: math.Inf(1)},
+			{Min: -1.5, Max: 1.5},
```

Comment:
I had to do 1.5 since sometimes we get the following value for `OZ: 1.0000000002` which causes for us to not be able to validate inputs on the base frame since it would be out of bounds if the value were set to `1`

Reply:
Add as a comment please

Reply:
I'm confused why would we be needing to check out of bounds numbers here?

Reply:
this isn't something that we would need to check in the `ptgKinematics.go` file, rather is it something we need to consider for in the `frame.go` file and the set-up for the thing we need to consider is here.
Specifically this is what we must account for:
```
// Interpolate interpolates the given amount between the two sets of inputs.
func (pf *poseFrame) Interpolate(from, to []Input, by float64) ([]Input, error) {
	if err := pf.baseFrame.validInputs(from); err != nil {
		return nil, NewIncorrectInputLengthError(len(from), 7)
	}
	if err := pf.baseFrame.validInputs(to); err != nil {
		return nil, NewIncorrectInputLengthError(len(to), 7)
	}
	return from, nil
}
```
It is possible for our `plan.Path()` object to have the following value: `x: someX, y:someY, z:someZ, OZ: OZ: 1.0000000002, Theta: someTheta`.
Hence, our frame's limits must bound the passed in value. If the values which we wish to interpolate are not bounded, then we error out here: `pf.baseFrame.validInputs(from)`.

Succinctly, If we were to initialize with `{Min: -1., Max: 1.}` we would not be able to interpolate `x: someX, y:someY, z:someZ, OZ: OZ: 1.0000000002, Theta: someTheta` which is a valid value.